File: SpiralMatrix.py

# Function to print a matrix in spiral order.
def getSpiralMatrix(matrix):
    top=0
    bottom=len(matrix)-1
    left=0
    right=len(matrix[0])-1
    while top<=bottom and left<=right:
        # traverse from left to right
        for col in range(left,right+1,1):
            print(matrix[top][col],end=" ")
        top+=1
        # traverse from top to bottom
        for row in range(top,bottom+1,1):
            print(matrix[row][right],end=" ")
        right-=1
        if top<=bottom:
            # traverse from right to left
            for col in range(right,left-1,-1):
                print(matrix[bottom][col],end=" ")
            bottom-=1
        if left<=right:
            #traverse from bottom to top
            for row in range(bottom,top-1,-1):
                print(matrix[row][left],end=" ")
            left+=1

# Let's Start here
matrix=[
    [1,2,3],
    [4,5,6],
    [7,8,9]
]
getSpiralMatrix(matrix)

File: 100-Days-Series/Day 44/SpiralMatrix.java
import java.util.ArrayList;
import java.util.List;
class SpiralMatrix{

    // Function to print a matrix in spiral order.
    public static List<Integer> getSpiralMatrix(int matrix[][]){
        List<Integer> result=new ArrayList<>();

        int top=0,bottom=matrix.length-1;
        int left=0,right=matrix[0].length-1;
        while(top<=bottom && left<=right){
            // traverse from left to right
            for(int col=left;col<=right;col++){
                result.add(matrix[top][col]);
            }
            top++;
            // traverse from top to bottom
            for(int row=top;row<=bottom;row++){
                result.add(matrix[row][right]);
            }
            right--;
            if(top<=bottom){
                // traverse from right to left
                for(int col=right;col>=left;col--){
                    result.add(matrix[bottom][col]);
                }
                bottom--;
            }
            if(left<=right){
                // traverse from bottom to top
                for(int row=bottom;row>=top;row--){
                    result.add(matrix[row][left]);
                }   
                left++;
            }
        }
        return result;
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start here
        int matrix[][]={
            {1,2,3},
            {4,5,6},
            {7,8,9}
        };
        List<Integer>spiralOrder=getSpiralMatrix(matrix);
        System.out.println("Spiral Order: " + spiralOrder);
    }
}